reloj-checador/app/controllers/attendaces_controller.rb

```diff
@@ -22,51 +22,58 @@ def edit
   # POST /attendaces or /attendaces.json
   def create
     @attendace = Attendace.new(attendace_params)
-    tim = Time.new
-    @attendace.date = tim.strftime("%Y-%m-%d")
-    @attendace.time = tim.strftime("%I:%M:%S")
-
-    respond_to do |format|
+    employee = Employee.find_by(private_number: attendace_params[:private_number])
+    if !employee.nil?
       if @attendace.save
-        format.html { redirect_to @attendace, notice: "Attendace was successfully created." }
-        format.json { render :show, status: :created, location: @attendace }
+        flash[:success] = "Employee #{employee.name} #{@attendace.check_type} registered"
+        redirect_to new_attendace_path
       else
-        format.html { render :new, status: :unprocessable_entity }
-        format.json { render json: @attendace.errors, status: :unprocessable_entity }
+        flash[:error] = @attendace.errors.full_messages
+        render "new"
       end
+    else
+      flash[:error] = "Private number not found"
+      render "new"
     end
   end
 
   # PATCH/PUT /attendaces/1 or /attendaces/1.json
   def update
-    respond_to do |format|
-      if @attendace.update(attendace_params)
-        format.html { redirect_to @attendace, notice: "Attendace was successfully updated." }
-        format.json { render :show, status: :ok, location: @attendace }
-      else
-        format.html { render :edit, status: :unprocessable_entity }
-        format.json { render json: @attendace.errors, status: :unprocessable_entity }
-      end
+    if @attendace.update(attendace_params)
+      flash[:success] = "Attendace was successfully updated."
+      redirect_to @attendaces_path
+    else
+      flash[:error] = @attendace.errors.full_messages
+      render @attendaces_path
     end
   end
 
   # DELETE /attendaces/1 or /attendaces/1.json
   def destroy
-    @attendace.destroy
-    respond_to do |format|
-      format.html { redirect_to attendaces_url, notice: "Attendace was successfully destroyed." }
-      format.json { head :no_content }
+    if @attendace.destroy
+      flash[:success] = "Record deleted"
+      render "index"
+    else
+      flash[:error] = @attendace.errors.full_messages
+      render "index"
```

Comment:
If the render "index" is part of the result of both options, I think it would be convenient to remove it from the if.

```
    if @attendace.destroy
      flash[:success] = "Record deleted"
    else
      flash[:error] = @attendace.errors.full_messages
    end
    render "index"
```

Reply:
Sugestion taken, and code changed. Code change also in update method for rendering and redirection.